runners/spark/src/main/java/org/apache/beam/runners/spark/coders/StatelessJavaSerializer.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.beam.runners.spark.coders;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.KryoException;
+import com.esotericsoftware.kryo.Serializer;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import com.esotericsoftware.kryo.util.ObjectMap;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+
+/**
+ * Stateless Java serializer.
+ * Needed due to a state re-use issue in Kryo version 2.21 used in Spark 1.x
+ * See: https://issues.apache.org/jira/browse/SPARK-7708
+ * See: https://github.com/EsotericSoftware/kryo/issues/312
+ * Copied from Kryo version 2.21.1 com/esotericsoftware/kryo/serializers/JavaSerializer.java
```

Comment:
R: @davorbonaci for license reasons - this was "copied" here because Spark uses Kryo 2.21, and even if we add dependency on 2.21.1 it will still break on clusters with a provided/pre-deployed Spark 1.x., so dependency won't work.
Is this OK in terms of license ? thanks.